### PR TITLE
ignore inherited properties in R.countBy

### PR DESCRIPTION
**Pull request by davidchambers**
Supersedes #758

I simplified the implementation while I was at it. :)

### CODE REVIEW

**Comment by davidchambers** [comment]
This pull request fixes a bug, so we should merge it and publish 0.9.1 before merging any other pull requests.

File: src/countBy.js [hunk 1]
```diff
@@ -25,8 +24,12 @@ var _keyValue = require('./internal/_keyValue');
...
-    }, {}, _keyValue(fn, list));
+    var counts = {};
+    var len = list.length;
+    var idx = -1;
+    while (++idx < len) {
+        var key = fn(list[idx]);
+        counts[key] = (has(key, counts) ? counts[key] : 0) + 1;
+    }
+    return counts;
 });
```
Comment (CrossEye) [inline src/countBy.js]: I wonder if we should do something similar to `sortBy` and eliminate the need for `_keyValue`.

Comment (davidchambers) [inline src/countBy.js]: Good idea. I've added a second commit to this pull request, which removes `_compareKeys` and `_keyValue`. :)

File: src/sortBy.js [hunk 2]
```diff
@@ -36,5 +34,9 @@ var _pluck = require('./internal/_pluck');
  *      sortByNameCaseInsensitive(people); //=> [alice, bob, clara]
  */
 module.exports = _curry2(function sortBy(fn, list) {
-    return _pluck('val', _keyValue(fn, list).sort(_compareKeys));
+    return clone(list).sort(function(a, b) {
+        var aa = fn(a);
+        var bb = fn(b);
+        return aa < bb ? -1 : aa > bb ? 1 : 0;
```
Comment (buzzdecafe) [inline src/sortBy.js]: can't you just `return aa - bb;`?

File: dist/ramda.js [hunk 3]
```diff
@@ -4250,7 +4162,11 @@
      *      sortByNameCaseInsensitive(people); //=> [alice, bob, clara]
      */
     var sortBy = _curry2(function sortBy(fn, list) {
-        return _pluck('val', _keyValue(fn, list).sort(_compareKeys));
+        return clone(list).sort(function (a, b) {
+            var aa = fn(a);
+            var bb = fn(b);
+            return aa < bb ? -1 : aa > bb ? 1 : 0;
```
Comment (buzzdecafe) [inline dist/ramda.js]: likewise

File: src/sortBy.js (hunk 2 above)
Comment (davidchambers) [inline src/sortBy.js]: Nope. We're comparing strings:

```javascript
> 'foo' - 'bar'
NaN
```

Comment (buzzdecafe) [inline src/sortBy.js]: i expected you would have a ready reply. LGTM

**Comment by CrossEye** [comment]
LVGTM

**Pull request merged**